exporter/instanaexporter/exporter.go

```diff
@@ -112,6 +115,32 @@ func (e *instanaExporter) export(ctx context.Context, url string, header map[str
 		return consumererror.NewPermanent(err)
 	}
 
+	if e.config.CAFile != "" {
+		caCertPool := x509.NewCertPool()
+
+		certData, readErr := os.ReadFile(e.config.CAFile)
+
+		if readErr != nil {
+			return consumererror.NewPermanent(readErr)
+		}
+
+		if ok := caCertPool.AppendCertsFromPEM(certData); !ok {
+			parsedCert, parseErr := x509.ParseCertificate(certData)
+
+			if parseErr != nil {
+				return consumererror.NewPermanent(parseErr)
+			}
+
+			caCertPool.AddCert(parsedCert)
+		}
```

Comment:
Do you normally return error when this is false `caCertPool.AppendCertsFromPEM(certData)`?

Reply:
Is it usually just PEM files?

Reply:
If `caCertPool.AppendCertsFromPEM(certData)` fails to parse at least one certificate from the series of PEM certificates, `ok` will return false, but not a "proper" error.

I am changing this code. `caCertPool.AppendCertsFromPEM(certData)` calls `x509.ParseCertificate(certData)` internally, so the code inside the condition is redundant.